packages/next/src/executors/build/lib/create-next-config-file.ts

```diff
@@ -76,7 +75,14 @@ export function createNextConfigFile(
     projectRoot
   );
   for (const moduleFile of moduleFilesToCopy) {
-    ensureDirSync(dirname(join(context.root, options.outputPath, moduleFile)));
+    const moduleFileDir = dirname(
+      join(context.root, options.outputPath, moduleFile)
+    );
+
+    if (!existsSync(moduleFileDir)) {
+      mkdirSync(moduleFileDir, { recursive: true });
+    }
```

Comment:
We don't need the `!existsSync(moduleFileDir)` check here, `mkdirSync` won't overwrite an existing directory.